### DISCUSSION TITLE
Convergence problem when using mortar based approach in contact calculation

### DISCUSSION THREAD

**Discussion by jin0123456** [idaholab/moose#27337] [Q&A Modules: General]
Hello

I have ased the similar question when my geometry is 2D RZ coordinate in https://github.com/idaholab/moose/discussions/27303.
I tried mortar approach in 3D situation but faced with convergence problem when it converged well in 2D simulation. There is no difference between two input files except mesh. There, https://github.com/jin0123456/miniature-system
When applied `[gap_contact]`, it would run in a few time steps then became not converged. When applied `[gap_contact1]`, it would not converge on the very first step.
So how to use 3D mortar based approach in the right way for my case?

**Comment by GiudGiud** [comment]
@recuero @lindsayad

**Comment by recuero** [comment]
Many things could be happening. One thing I'd change right away is your solver settings. For example, to (Newton could work too):

```
  type = Transient
  solve_type = 'PJFNK'

  petsc_options = '-snes_ksp_ew'
  petsc_options_iname = '-pc_type -pc_factor_mat_solver_type -pc_factor_shift_type -pc_factor_shift_amount -mat_mffd_err'
  petsc_options_value = 'lu       superlu_dist                  NONZERO               1e-15                   1e-5'
```

I have run this type of problems in 3D, there shouldn't be a major problem. Also make sure sure you build with the appropriate automatic differentiation size, which I assume you are doing since you didn't get a error message to that effect.

**Reply by jin0123456** [reply]
Thanks for your advice. As many things could happen, I will try to find out the reason.
